src/app/public/modules/colorpicker/colorpicker.component.spec.ts

```diff
@@ -186,443 +211,640 @@ describe('Colorpicker Component', () => {
+    }));
+
+    it('should allow user to click apply the color change.', fakeAsync(() => {
+      component.selectedOutputFormat = 'hex';
+      openColorpicker(nativeElement, fixture);
+      setInputElementValue(nativeElement, 'hex', '#2B7230');
+      applyColorpicker(nativeElement, fixture);
+      verifyColorpicker(nativeElement, '#2b7230', '43, 114, 48');
+    }));
+
+    it('should reset colorpicker via reset button.', fakeAsync(() => {
+      fixture.detectChanges();
+      let spyOnResetColorPicker = spyOn(colorpickerComponent, 'resetPickerColor').and.callThrough();
+      fixture.detectChanges();
+      tick();
+      openColorpicker(nativeElement, fixture);
+      setPresetColor(nativeElement, fixture, 4);
+      fixture.detectChanges();
+      tick();
+      const buttonElem = nativeElement.querySelector('.sky-colorpicker-reset-button') as HTMLElement;
+      buttonElem.click();
+      tick();
+      fixture.detectChanges();
+      expect(spyOnResetColorPicker).toHaveBeenCalled();
+      verifyColorpicker(nativeElement, 'rgba(40,137,229,1)', '40, 137, 229');
+    }));
+
+    it('should accept open colorpicker via messageStream.', fakeAsync(() => {
+      fixture.detectChanges();
+      tick();
+      component.sendMessage(SkyColorpickerMessageType.Open);
+      tick();
+      fixture.detectChanges();
+      tick();
+      verifyMenuVisibility();
+    }));
+
+    it('should accept reset colorpicker via messageStream.', fakeAsync(() => {
+      component.colorControl.setValue('rgba(40,137,229,1)');
+      fixture.detectChanges();
+      tick();
+      fixture.detectChanges();
+      verifyColorpicker(nativeElement, 'rgba(40,137,229,1)', '40, 137, 229');
+      tick();
+      openColorpicker(nativeElement, fixture);
+      setPresetColor(nativeElement, fixture, 4);
+      fixture.detectChanges();
+      tick();
+      component.sendMessage(SkyColorpickerMessageType.Reset);
+      tick();
+      fixture.detectChanges();
+      tick();
+      verifyColorpicker(nativeElement, 'rgba(40,137,229,1)', '40, 137, 229');
+    }));
+
+    it('should toggle reset button via messageStream.', fakeAsync(() => {
+      fixture.detectChanges();
+      tick();
+      expect(nativeElement.querySelectorAll('.sky-colorpicker-reset-button').length).toEqual(1);
+      component.sendMessage(SkyColorpickerMessageType.ToggleResetButton);
+      tick();
+      fixture.detectChanges();
+      tick();
+      expect(nativeElement.querySelectorAll('.sky-colorpicker-reset-button').length).toEqual(0);
+      component.sendMessage(SkyColorpickerMessageType.ToggleResetButton);
+      tick();
+      fixture.detectChanges();
+      tick();
+      expect(nativeElement.querySelectorAll('.sky-colorpicker-reset-button').length).toEqual(1);
+    }));
+
+    it('should only emit the form control valueChanged event once per change', async(() => {
+      fixture.detectChanges();
+      let count = 0;
+      component.colorForm.valueChanges.subscribe(() => {
```

Comment:
Since this is an async test, have you double checked to make sure its waiting until completion to do all `valueChanges` events are fired? Also, why not just use a `spy` here?

Reply:
I didn't use a spy due to the fact that I need to wait for all async things to be done. A spy might deliver a false result on the initial expectation. I need some way to asynchronously ensure that everything has completed before marking the tests as passed. 

I did verify that if I remove my changes this test fails due to `valueChanges` firing more times.

Reply:
Ok good! I had this same issue with an async test last week. I solved it with a setTimeout, but maybe this is the better approach.

Well if you've confirmed it, then I suppose this looks good to me. 👍

Reply:
I'm inclined to agree with Alex. I also think we should have an `expect` in here somewhere.

Reply:
Done